### PR TITLE
Register report view

### PR DESCRIPTION
**Pull request by paulinembabu**
See Issue https://github.com/OpenSRP/opensrp-client-emmunize-malawi/issues/3

### CODE REVIEW

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        ListView listView = findViewById(R.id.lv_reportRegister_groupings);
+        TextView titleTv = findViewById(R.id.title);
+
+        if (titleTv != null) {
+            titleTv.setText(R.string.reports_type);
+        }
```
Comment (bennsimon) [inline opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:26]: you can move this to a `setUpViews()` method that will then be inherited by this class

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                case 1:
+                case 2:
+                    Intent intent = new Intent(ReportTypeListRegisterActivity.this, ReportRegisterActivity.class);
+                    startActivity(intent);
+                    break;
+                default:
+                    break;
+            }
+
+        });
```
Comment (bennsimon) [inline opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:43]: you can move this to a `loadData()` method that will then be inherited by this class and overridden

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public void onSyncStart() {
+        // Nothing to happen here
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        createDrawer();
+    }
```
Comment (bennsimon) [inline opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:67]: Is this needed?

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.giz.R;
+import org.smartregister.giz.model.ReportGroupingModel;
+
+import java.util.ArrayList;
+
+public class ReportTypeListRegisterActivity extends ReportRegisterActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (bennsimon) [inline opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:18]: There is no need to override this method.

**Review by bennsimon — CHANGES_REQUESTED** [review]

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java (hunk 4 above)
Comment (bennsimon) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:18]: Rather override this https://github.com/OpenSRP/opensrp-client-giz-malawi/pull/358#discussion_r570155823

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java (hunk 3 above)
Comment (paulinembabu) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:67]: yes

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java (hunk 1 above)
Comment (paulinembabu) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java]: Done

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java (hunk 2 above)
Comment (paulinembabu) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java:38]: Done

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java (hunk 3 above)
Comment (paulinembabu) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java]: Removed

File: opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java (hunk 4 above)
Comment (paulinembabu) [reply opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/ReportTypeListRegisterActivity.java]: Done

**Review by bennsimon — APPROVED** [review]

**Pull request merged**